components/brave_sync/brave_profile_sync_service_impl.h

```diff
@@ -215,6 +221,13 @@ class BraveProfileSyncServiceImpl
 
   bool IsSQSReady() const;
 
+  // Methods to run specified actions right after bookmark model was loaded
+  void OnSetupSyncNewToSyncBkmLoaded();
+  void OnSetupSyncHaveCodeBkmLoaded(const Uint8Array& seed);
```

Comment:
we shouldn't queue `OnSetupSyncNewToSync` and `OnSetupSyncHaveCode` because both have to wait `OnSyncReady` so we only need to queue up `OnSyncReady``